### PR TITLE
Feature: Update merchant to be empty and with validation 

### PR DESCRIPTION
**Pull request by waterim**
### Details

This PR is about to add an error message when merchant is empty on expense reports in request money and split bill
### Fixed Issues

$ https://github.com/Expensify/App/issues/32222

### Tests

1. Login to the app
2. Click on "Request money" or "Split bill"
3. Select an amount
**For IOU request**:
4. Select user, not a workspace
5. Merchant should be empty and no validation errors 
6. You are able to create a request with empty merchant
7. Create request
8. Go to edit page
9. No (empty) string visible for the merchant
10. Try to edit with empty string the merchant
11. You were able to edit with empty string
**For Expense request**:
4. Select a workspace
5. Look that merchant is empty, but validation error is appear that you need to provide a merchant
6. Create request
7. Go to edit page
8. try to edit merchant as empty field
9. validation error will appear

———-

IOU 
1. Global Create + >  “Request money”
2. Enter an amount 
3. Choose an individual 
4. Verify the merchant is empty on the confirmation page
5. Click confirm to create the request 
6. Verify you aren’t prompted to enter a merchant name to proceed with creating the request
7. Click the report preview component to navigate to the expense report 
8. Verify the request preview component doesn’t contain a “faux” merchant (i.e none, unknown merchant etc)
9. Click the request preview component to navigate to the transaction thread
10. Verify the merchant field is empty 
11. Click into the merchant field 
12. Verify the merchant edit field is empty 
13. Enter a merchant name
14. Verify you were able to enter a merchant 
15. Navigate back to the expense report view  
16. Verify the request preview component contains the merchant name entered
- [x] Verify that no errors appear in the JS console

### Offline tests

N/A automated tests

### QA Steps

Same as tests

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [x] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>Web</summary>

image

</details>

### CODE REVIEW

File: src/components/MoneyRequestConfirmationList.js [hunk 1]
```diff
@@ -690,7 +692,11 @@ function MoneyRequestConfirmationList(props) {
                             disabled={didConfirm}
                             interactive={!props.isReadOnly}
                             brickRoadIndicator={shouldDisplayFieldError && TransactionUtils.isMerchantMissing(transaction) ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
-                            error={shouldDisplayFieldError && TransactionUtils.isMerchantMissing(transaction) ? translate('common.error.enterMerchant') : ''}
+                            error={
+                                (props.isPolicyExpenseChat && !props.iouMerchant) || (shouldDisplayFieldError && TransactionUtils.isMerchantMissing(transaction))
```
[the remark this answers is not in the document]
Comment (rezkiy37) [reply src/components/MoneyRequestConfirmationList.js:696]: Moreover, I would like to extract all this logic out of JSX. Let's create a variable like `hasMerchantError`

**Review by rezkiy37 — COMMENTED** [review]
Left comments.

**Review by mountiny — CHANGES_REQUESTED** [review]
@waterim I think there was one condition not clear and that is when you are using SmartScan so submitting a picture receipt we still do not require a merchant and it should work the same as it does now and set the `(none)` partial merchant optimistically

**Comment by waterim** [comment]
@mountiny Okay, pushed changes, removed required for isScanRequest expense report, now it works as before my changes.
If I understand you correctly only this required in smart scan was an issue, right?

File: src/components/MoneyRequestConfirmationList.js [hunk 2]
```diff
@@ -267,6 +267,8 @@ function MoneyRequestConfirmationList(props) {
         return (props.hasSmartScanFailed && TransactionUtils.hasMissingSmartscanFields(transaction)) || (didConfirmSplit && TransactionUtils.areRequiredFieldsEmpty(transaction));
     }, [props.isEditingSplitBill, props.hasSmartScanFailed, transaction, didConfirmSplit]);
 
+    const isIOUMerchantPresent = props.isPolicyExpenseChat && !props.isScanRequest && !props.iouMerchant;
```
Comment (mountiny) [inline src/components/MoneyRequestConfirmationList.js:270]:
```suggestion
    const isMerchantEmpty = props.isPolicyExpenseChat && !props.isScanRequest && !props.iouMerchant;
```

File: src/libs/actions/IOU.js [hunk 3]
```diff
@@ -101,7 +101,7 @@ function resetMoneyRequestInfo(id = '') {
         currency: lodashGet(currentUserPersonalDetails, 'localCurrencyCode', CONST.CURRENCY.USD),
         comment: '',
         participants: [],
-        merchant: CONST.TRANSACTION.DEFAULT_MERCHANT,
+        merchant: '',
```
Comment (mountiny) [inline src/libs/actions/IOU.js:104]: this will not set the merchant as `Request` default for IOU requests, right?

**Review by mountiny — COMMENTED** [review]
@waterim thanks, getting closer, is this the cleanest way to add the merchant empty check to this form? also wont the reset of the money request remove the `Request` from merchant too?

File: src/libs/actions/IOU.js (hunk 3 above)
Comment (waterim) [reply src/libs/actions/IOU.js:104]: Yes, as I understand from the ticket we need to make it clear, not?

File: src/components/MoneyRequestConfirmationList.js (hunk 2 above)
Comment (waterim) [reply src/components/MoneyRequestConfirmationList.js:270]: "is this the cleanest way to add the merchant empty check to this form?"
I think yes, we need to check for an expense chat, but scanRequest should work as previously.

"also wont the reset of the money request remove the Request from merchant too?"
Hm, didn't get a question, don't we want to remove default "Request" text from the merchant?

Comment (waterim) [reply src/components/MoneyRequestConfirmationList.js:270]: Ahh, okay, you mean that merchant field must be empty only for expense chats, correct?
@mountiny

Comment (mountiny) [reply src/components/MoneyRequestConfirmationList.js:270]: We sont want to remove it for IOU peer to peer request. Only for expense requests so group policies

File: src/libs/actions/IOU.js (hunk 3 above)
Comment (mountiny) [reply src/libs/actions/IOU.js]: For IOUs it stays `Request` though still I think?

Comment (waterim) [reply src/libs/actions/IOU.js]: We agreed on (none), no? in the conversation

Comment (waterim) [reply src/libs/actions/IOU.js]: here is an outdated code, I changed reset to (none)

File: src/components/MoneyRequestConfirmationList.js [hunk 4]
```diff
@@ -267,6 +267,9 @@ function MoneyRequestConfirmationList(props) {
         return (props.hasSmartScanFailed && TransactionUtils.hasMissingSmartscanFields(transaction)) || (didConfirmSplit && TransactionUtils.areRequiredFieldsEmpty(transaction));
     }, [props.isEditingSplitBill, props.hasSmartScanFailed, transaction, didConfirmSplit]);
 
+    const isMerchantEmpty = !props.iouMerchant || props.iouMerchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT || props.iouMerchant === CONST.TRANSACTION.DEFAULT_MERCHANT;
```
Comment (mountiny) [inline src/components/MoneyRequestConfirmationList.js:270]: We can allow them to leave the merchant `Request`, we will not default to it now I would say though for expense reports so they have to fill stuff out
```suggestion
    const isMerchantEmpty = !props.iouMerchant || props.iouMerchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT;
```

File: src/pages/EditRequestMerchantPage.js [hunk 5]
```diff
@@ -18,22 +18,27 @@ const propTypes = {
...
-        if (_.isEmpty(value.merchant)) {
-            errors.merchant = 'common.error.fieldRequired';
-        }
-
-        return errors;
-    }, []);
+    const validate = useCallback(
+        (value) => {
+            const errors = {};
+            if (_.isEmpty(value.merchant) && isPolicyExpenseChat) {
```
Comment (mountiny) [inline src/pages/EditRequestMerchantPage.js:35]: Should we also trim the value here? so they cannot put in `   ` whitespaces as merchant and be considered as valid??

File: src/pages/iou/MoneyRequestMerchantPage.js [hunk 6]
```diff
@@ -53,6 +53,7 @@ function MoneyRequestMerchantPage({iou, route}) {
     const {inputCallbackRef} = useAutoFocusInput();
     const iouType = lodashGet(route, 'params.iouType', '');
     const reportID = lodashGet(route, 'params.reportID', '');
+    const isEmptyMerchant = iou.merchant === '' || iou.merchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT || iou.merchant === CONST.TRANSACTION.DEFAULT_MERCHANT;
```
Comment (mountiny) [inline src/pages/iou/MoneyRequestMerchantPage.js:56]: Same as above `Request` is not empty merchant I think
```suggestion
    const isEmptyMerchant = iou.merchant === '' || iou.merchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT;
```

**Review by mountiny — CHANGES_REQUESTED** [review]

**Comment by mountiny** [comment]
I ma making a test build for easier testing

**Review by mountiny — CHANGES_REQUESTED** [review]
BUG: Requesting manual money from group policy is possible without merchant, the field should be required and it should not be hidden behind the Show More 
image

There is also some weird bug here, which is also repro in staging, however, from the video you can see the merchant is set to `Request` for a second which seems to be a bug in the optimistic data as the merchant should be `(none)` in this case and hence it should show as empty

https://github.com/Expensify/App/assets/36083550/538597e0-dc52-4b19-9814-e7b588453807

**Comment by waterim** [comment]
@mountiny Damn, thats super weird, this happens after I merged main into this branch..

**Comment by waterim** [comment]
I cant even see console.logs in MoneyRequestConfirmationList.js component..

**Comment by waterim** [comment]
Okay, I found, there was some refactor and all my code is unused now.. :D
image

**Comment by mountiny** [comment]
Discussing in slack

**Comment by waterim** [comment]
@mountiny updated new code(hopefully everything), tested and works fine

**Comment by waterim** [comment]
Some weird TS issue unrelated to my code

**Comment by waterim** [comment]
@mountiny merged main again, lint TS issues was resolved, but some test which is unrelated is failing
And for some reason tests are running for so long, 10 mins so far :D

image

**Comment by mountiny** [comment]
@waterim the rundown of the tests with @trjExpensify has been successful only updated was required to the design of the required field which should he been addressed already I believe.

Once the conflicts are resolved we can proceed with the PR

**Comment by koko57** [comment]
@mountiny conflicts resolved

**Review by mountiny — APPROVED** [review]
Code is looking good to me

**Comment by mountiny** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by mountiny** [comment]
@koko57 @waterim 

Been testing around and noticed two issues:
- the merchant validation should only happen on submission according to our guidelines. When you are creating the request, the merchant fields should not show as required. the error should pop up only once the form has been submitted and the required merchant field is empty
- On the split bill screen or in Scan request the merchant field shows up as required, but then it disappears as correctly it should not be there when scanning in workspace as we are running smartscan

https://github.com/Expensify/App/assets/36083550/710b532b-b69f-4e7c-8357-c4f8fb62efcb

**Review by mountiny — CHANGES_REQUESTED** [review]
Couple notes in comment above

**Comment by koko57** [comment]
@mountiny I think I fixed the first issue, will push the changes in a minute. I just wanted to ask - if we fill in the merchant field it should disappear (I mean move to the hidden fields under Show more button)?

https://github.com/Expensify/App/assets/28020445/67a1c96f-5729-4e0f-a5ea-c3ab0ae1f732

It's a current behaviour, I changed it in my commit and now it's visible for workspace request. Let me know if I should restore hiding of this field.

For the second issue - I don't really understand, could you explain what is the expected behavior here? And is it on recording? In which second it starts?

**Comment by mountiny** [comment]
image

the issue if that the merchant should not even be visible for receipt scan creation. We run smartscan which should fill in the merchant so its not useful to have it in there. 

In the video you could see that the merchant was showing in the scan flow which is incorrect

> if we fill in the merchant field it should disappear (I mean move to the hidden fields under Show more button)?

It should not disappear, for manual requests for workspaces, the merchant is required so it should be visible at all times

The bug there is that the field is showing as required when you navigate to the page immediately, thats not how we handle required fields, it should start showing as required only once the user click the button to proceed and the merchant is missing/ validation fails

**Comment by koko57** [comment]
> It should not disappear, for manual requests for workspaces, the merchant is required so it should be visible at all times

so in my last commit, I applied this change for this field to be always visible 

> The bug there is that the field is showing as required when you navigate to the page immediately, thats not how we handle required fields, it should start showing as required only once the user click the button to proceed and the merchant is missing/ validation fails

yes, now it works that way.

Thanks for explaining the issue with the Scan request, I'll take care of it 🙂

**Comment by koko57** [comment]
@mountiny for scans - it should be fixed now

**Review by mountiny — APPROVED** [review]
Tested new changes and it works well, thank you very much!

File: src/languages/en.ts [hunk 7]
```diff
@@ -618,6 +618,7 @@ export default {
             genericSmartscanFailureMessage: 'Transaction is missing fields',
             atLeastTwoDifferentWaypoints: 'Please enter at least two different addresses',
             splitBillMultipleParticipantsErrorMessage: 'Split bill is only allowed between a single workspace or individual users. Please update your selection.',
+            invalidMerchant: 'Please enter a corrent merchant.',
```
Comment (situchan) [inline src/languages/en.ts:621]: typo - #33781

**Comment by situchan** [comment]
Coming from https://github.com/Expensify/App/issues/33786:
Manual request with scan receipt case was not considered.

File: src/components/MoneyRequestConfirmationList.js [hunk 8]
```diff
@@ -500,7 +503,7 @@ function MoneyRequestConfirmationList(props) {
         }
 
         const shouldShowSettlementButton = props.iouType === CONST.IOU.TYPE.SEND;
-        const shouldDisableButton = selectedParticipants.length === 0;
+        const shouldDisableButton = selectedParticipants.length === 0 || shouldDisplayMerchantError;
```
Comment (jayeshmangwani) [inline src/components/MoneyRequestConfirmationList.js:506]: Coming from https://github.com/Expensify/App/issues/38130:

Form Submit buttons should not be disabled or blocked from being pressed

**Pull request merged**